src/mozilla_sec_eia/extract.py

```diff
@@ -0,0 +1,244 @@
+        run_id = most_recent_run.info.run_id
+
+    filings_to_extract = metadata[~metadata["filename"].isin(extraction_metadata.index)]
+    if num_filings > 0:
+        filings_to_extract = filings_to_extract.sample(num_filings)
+    return (
+        filings_to_extract,
+        extraction_metadata,
+        extracted,
+        run_id,
+    )
+
+
+def _get_experiment_name(dataset: str, experiment_suffix: str | None = None) -> str:
+    experiment_name = f"{dataset}_extraction"
+    if experiment_suffix is not None:
+        experiment_name += f"_{experiment_suffix}"
+    return experiment_name
+
+
+def compute_validation_metrics(
+    computed_set: pd.DataFrame,
+    validation_set: pd.DataFrame,
+    value_col: str,
+) -> dict:
+    """Compute precision and recall for extraction compared to validation set.
+
+    Arg:
+        computed_set: Extracted data.
+        validation_set: Expected extraction results.
+        value_col: Column to compare when computing metrics.
+    """
+    # Get initial length of both sets
+    computed_len = len(computed_set)
+    validation_len = len(validation_set)
+
+    # Get index of rows only in one set and make Null in other set
+    idx_validation_only = validation_set.index.difference(computed_set.index)
+    padded_compute_set = pd.concat(
+        [
+            computed_set[value_col],
+            pd.Series([None] * len(idx_validation_only), index=idx_validation_only),
+        ]
+    ).sort_index()
+    idx_compute_only = computed_set.index.difference(validation_set.index)
+    padded_validation_set = pd.concat(
+        [
+            validation_set[value_col],
+            pd.Series([None] * len(idx_compute_only), index=idx_compute_only),
+        ]
+    ).sort_index()
+
+    true_positives = (padded_compute_set == padded_validation_set).sum()
+
+    return {
+        "precision": true_positives / computed_len,
+        "recall": true_positives / validation_len,
+    }
+
+
+def validate_extraction(dataset: str):
+    """Run extraction on validation set and compare results to labeled data."""
+    validation_set = pd.read_csv(
+        resources.files("mozilla_sec_eia.package_data") / f"{dataset}_labels.csv"
+    )
+
+    # Get metadata for labelled filings
+    archive = GCSArchive()
+    to_extract = archive.get_metadata(filenames=list(validation_set["filename"]))
+
+    # Extract data from filings
+    extracted = extract_filings(
+        dataset=dataset, metadata=to_extract, experiment_suffix="validation"
+    )
+    # Set index for validation set based on returned extracted DF
+    validation_set = validation_set.set_index(extracted.index.names)
+
+    # Get extraction run from mlflow and start again to log validation metrics
+    experiment_name = _get_experiment_name(dataset, experiment_suffix="validation")
+    run = _get_most_recent_run(experiment_name)
+    with mlflow.start_run(run_id=run.info.run_id):
+        # Compute metrics and log
+        if dataset == "basic_10k":
```

Comment:
Is there any reason to confine this to basic 10K or was it just because this framework isn't implemented in the Ex. 21 extraction yet? I guess we'll probably have different validation metrics for each dataset once the Ex. 21 validation is implemented. I think you could just take this `if` out for now.

Reply:
I figured we'd have a separate block for running the exhibit 21 extraction, so I was trying to set up for that. At the time being it doesn't really make much difference though